Platformio/HAL/HardwareAbstract.hpp

```diff
@@ -40,23 +29,17 @@ class HardwareAbstract {
   /// @param onBatteryStatusChangeHandler - Callable to be ran when batter status changes
   void onBatteryChange(std::function<void(batteryStatus)> onBatteryStatusChangeHandler);
   
-
+  virtual std::shared_ptr<wifiHandlerInterface> wifi() = 0;
```

Comment:
I do like the idea of having this but I am curious if we are going to force the downstream to check if the return if null?